sdk/src/test/scala/com/horizen/block/MainchainBlockReferenceTest.scala

```diff
@@ -248,4 +254,326 @@ class MainchainBlockReferenceTest extends JUnitSuite {
+    assertEquals("Block equihash solution length is wrong.", params1.EquihashSolutionLength, mcblock1.header.solution.length)
+
+    //Check for sidechain 2
+    val scIdHex2 = "00000000000000000000000000000000000000000000000000000000deadbeed"
+    val scId2 = new ByteArrayWrapper(BytesUtils.fromHexString(scIdHex2))
+
+    val params2 = RegTestParams(scId2.data)
+
+    val mcblockTry2 = MainchainBlockReference.create(mcBlockBytes, params2)
+
+    assertTrue("Block expected to be parsed", mcblockTry2.isSuccess)
+    val mcblock2 = mcblockTry2.get
+
+    assertTrue("Block expected to be semantically valid", mcblock2.semanticValidity(params2))
+
+    assertTrue("Block must contain transaction.", mcblock2.sidechainRelatedAggregatedTransaction.isDefined)
+    assertTrue("Block must not contain certificate.", mcblock2.backwardTransferCertificate.isEmpty)
+    assertTrue("Block must contain proof.", mcblock2.mproof.isDefined)
+    assertTrue("Block must not contain proof for left neighbor.", mcblock2.proofOfNoData._1.isEmpty)
+    assertTrue("Block must not contain proof for right neighbor.", mcblock2.proofOfNoData._2.isEmpty)
+
+    //Check sidechain creation transaction.
+    val aggTx2 = mcblock2.sidechainRelatedAggregatedTransaction.get
+    val outputs2 = aggTx2.mc2scTransactionsOutputs()
+    assertEquals("MC2SCAggTx unlockers size is different", 0, aggTx2.unlockers().size())
+    assertEquals("MC2SCAggTx outputs size is different", 1,outputs2.size())
+
+    //Check for sidechain 3
+    val scIdHex3 = "00000000000000000000000000000000000000000000000000000000deadbeef"
+    val scId3 = new ByteArrayWrapper(BytesUtils.fromHexString(scIdHex3))
+
+    val params3 = RegTestParams(scId3.data)
+
+    val mcblockTry3 = MainchainBlockReference.create(mcBlockBytes, params3)
+
+    assertTrue("Block expected to be parsed", mcblockTry3.isSuccess)
+    val mcblock3 = mcblockTry3.get
+
+    assertTrue("Block expected to be semantically valid", mcblock3.semanticValidity(params3))
+
+    assertTrue("Block must contain transaction.", mcblock3.sidechainRelatedAggregatedTransaction.isDefined)
+    assertTrue("Block must not contain certificate.", mcblock3.backwardTransferCertificate.isEmpty)
+    assertTrue("Block must contain proof.", mcblock3.mproof.isDefined)
+    assertTrue("Block must not contain proof for left neighbor.", mcblock3.proofOfNoData._1.isEmpty)
+    assertTrue("Block must not contain proof for right neighbor.", mcblock3.proofOfNoData._2.isEmpty)
+
+    //Check sidechain creation transaction.
+    val aggTx3 = mcblock3.sidechainRelatedAggregatedTransaction.get
+    val outputs3 = aggTx3.mc2scTransactionsOutputs()
+    assertEquals("MC2SCAggTx unlockers size is different", 0, aggTx3.unlockers().size())
+    assertEquals("MC2SCAggTx outputs size is different", 1,outputs3.size())
+
+  }
+
+  @Test
+  def blockForwardTransfer2Sidechains(): Unit = {
```

Comment:
please, add description for each Test. It's not easy to understand what is inside